browser/resources/brave_new_tab_page_refresh/api/background_impl.ts

```diff
@@ -101,6 +102,9 @@ export function initializeBackgrounds(
     },
 
     selectBackground(type, value) {
+      if (!value && type !== SelectedBackgroundType.kBrave) {
+        resetRandomizer()
+      }
```

Comment:
this naturally feels like a React.useMemo

Reply:
Changes to "current background" have been removed from this PR and will be addressed in https://github.com/brave/brave-browser/issues/46034.